osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/Components/SliderTailPiece.cs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Caching;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Input;
+using osu.Framework.Input.Events;
+using osu.Framework.Utils;
+using osu.Game.Graphics;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Screens.Edit;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Input;
+
+namespace osu.Game.Rulesets.Osu.Edit.Blueprints.Sliders.Components
+{
+    public partial class SliderTailPiece : SliderCircleOverlay
+    {
+        /// <summary>
+        /// Whether this is currently being dragged.
+        /// </summary>
+        private bool isDragging;
+
+        private InputManager inputManager = null!;
+
+        private readonly Cached<SliderPath> fullPathCache = new Cached<SliderPath>();
+
+        [Resolved(CanBeNull = true)]
```

Comment:
```suggestion
        [Resolved]
```

The nullability annotation on `EditorBeatmap` already implies the "can be null" part.